src/pages/User.tsx

```diff
@@ -57,6 +69,33 @@ const UserPage = observer(() => {
     triggerDownload(csvString, `${getExportFileName()}.csv`, 'text/csv');
   };
 
+  const DXD = useContract(
+    configStore.getNetworkContracts().votingMachines.dxd.token,
+    ERC20Json.abi
+  );
+
+  const updateUserVestingContracts = () => {
+    const contracts = Promise.all(
+      userVestingContracts.map(async contract => {
+        try {
+          const balance = await DXD?.balanceOf(contract.address);
+          return {
+            ...contract,
+            value: bnum(balance),
+          };
+        } catch (e) {
+          return {
+            ...contract,
+            value: bnum('0'),
+          };
+        }
+      })
+    );
+    contracts.then(setTokenVestingContracts);
+  };
+
+  useEffect(updateUserVestingContracts, []); // eslint-disable-line

```

Comment:
I think we should have `userAddress ` in the brackets of the useEffect so when the user goes from one profile to another it updates the contracts. Currently it does not

Reply:
Yes, great catch! Done